src/gcp/cloudsql/permissions_setup.ts

```diff
@@ -114,20 +117,36 @@ export async function setupSQLPermissions(
   }
   await setupIAMUsers(instanceId, databaseId, options);
 
+  let runGreenfieldSetup = false;
   if (schemaInfo.setupStatus === SchemaSetupStatus.GreenField) {
+    runGreenfieldSetup = true;
     logger.info(
-      `Database ${databaseId} has already been setup. Rerunning setup to repair any missing permissions.`,
+      `Database ${databaseId} has already been setup as Greenfield project. Rerunning setup to repair any missing permissions.`,
     );
-    await greenFieldSchemaSetup(instanceId, databaseId, schema, options, silent);
-    return SchemaSetupStatus.GreenField;
-  } else {
-    logger.info(`Detected schema "${schema}" setup status is ${schemaInfo.setupStatus}.`);
   }
 
-  // We need to setup the database
   if (schemaInfo.tables.length === 0) {
+    runGreenfieldSetup = true;
```

Comment:
Nit: We have inconsistent capitalization on greenfield and brownfield (some places it's greenField, some it's greenfield). Wikipedia says 'greenfield' (https://en.wikipedia.org/wiki/Greenfield_project), so let's settle on that